### ISSUE TITLE
INFECTOR's learning parameters T and New initiators in test set

### ISSUE THREAD

**Issue by FeiGSSS** [GiorgosPanagopoulos/Influence-Maximization-via-Representation-Learning#2]
Hi,
I have two confusion about your framework:

1.  the T is a parameters matrix to learn, right? if E=50, then the number of parameters here is 50x1170689 for Sina Weibo!? Is it too many?

2. You just trained the probabilities of initiators in the training set, how about that in the test set? Just ignore the initiators in test set which does not initialize a cascade in training set?

**Comment by geopanag** [comment]
Hello,
1. Please bear in mind that the pc I run the experiments on has 250Gb Ram. T is not really big, but D is (initiators x nodes), which is why I do a lot of np.around at iminfector.py.

2. Correct, the initiators in the test set have not started a cascade in the training set and are thus overlooked.

**Comment by FeiGSSS** [comment]
Hi,
Thx for your reply!

1. Yes,  D is much bigger and you can make some compromise to fit it in memory. But another concern arises about the calculation in eq. (2): the summation consists of N positive parts, which may cause a computational overflow in a big graph?  Is it possible when dealing with your data?

2. So, have you ever considered the intersection of the initiators in the test set and train set? I mean that if the intersection is empty, then you can't predict any cascade in the test set. Your work is based on an assumption: all initiators in the test set are also the initiators in the train set.

**Comment by geopanag** [comment]
Thanks for the questions.
1.  For eq. (2) specifically, we mention that we employ noise contrastive estimation to compute the summation.
2. The intersection is never empty because the test set has always a significant time span relative to the dataset (e.g. a twitter influencer will tweet at least once in a month). 
The test initiators that are not present in the train set cannot be predicted by diffusion methods but can be predicted by network methods such as IMM.
Overall DNI evaluation is not perfect (as we mention in the text "Although not devoid of assumptions...") but can be considered more realistic than simulating simple diffusion models. Moreover, that is exactly why we compute the precision as a supplementary evaluation. The idea is that a method should be able to extract seeds that are not only influential but also create cascades frequently, so they will be present in the test set.
